src/Controller/CommonAjaxController.php

```diff
@@ -0,0 +1,363 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller;
+
+use CommonDBTM;
+use Glpi\Http\Response;
+
+/**
+ * Ajax controller for handling "update", "delete", "restore" and "purge"
+ * actions on CommonDBTM objects
+ */
+class CommonAjaxController
+{
+    /**
+     * Object to update, delete, restore or purge
+     * @var CommonDBTM
+     */
+    protected CommonDBTM $item;
+
+    /**
+     * Handle a given POST request
+     *
+     * @param array $post POST data
+     *
+     * @return Response
+     */
+    final public function handleRequest(array $post): Response
```

Comment:
I gues your controller may also handle read oprations. You should not care whether data is sent in POST or GET. 
```suggestion
    final public function handleRequest(array $input): Response
```